src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -553,16 +553,16 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
         if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
             $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
 
-            $legacyTranslationsDir = dirname($r->getFilename()).'/../../Resources/translations';
+            // with Symfony 2.4, the Security component was split into several subpackages
+            // and the translations have been moved to the symfony/security-core package
+            $translationsDir = dirname($r->getFilename()).'/../Resources/translations';
 
-            if (file_exists($legacyTranslationsDir)) {
+            if (!file_exists($translationsDir) && file_exists($dir = dirname($r->getFilename()).'/../../Resources/translations')) {
```

Comment:
wait, we have the new path in 2.3 with only incomplete files ? This looks wrong to me.